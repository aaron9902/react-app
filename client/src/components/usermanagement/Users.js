import React, { Fragment, useEffect, useState } from 'react';
import UserItem from './UserItem';
import axios from 'axios';

const Users = () => {
    const [users, setUsers] = useState([]); 

// Get all users and update the users state
    useEffect(() => {
        let isMounted = true; 
        axios.get('/api/users/').then((res) => { 
            if (isMounted) setUsers(res.data); 
        })
        return () => { isMounted = false };
    }, []);

    return (
        <Fragment>
            {users.map(user => (
                <UserItem key={user._id} user={user} />
            ))}
        </Fragment>
    )
}

export default Users;