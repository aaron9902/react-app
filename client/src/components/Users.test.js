import React from 'react';
import { mount } from 'enzyme';
import Users from './usermanagement/Users';
import { BrowserRouter, BrowserRouter as Router, Link } from "react-router-dom"; 

describe('<Users/ >', () => {

  
//StepID: S500
  it('matches snapshot', () => {
    const wrapper = mount(<BrowserRouter><Users /></BrowserRouter>);
    expect(wrapper).toMatchSnapshot();
  });


});
