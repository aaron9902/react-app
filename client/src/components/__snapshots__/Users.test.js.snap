// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users/ > matches snapshot 1`] = `ReactWrapper {}`;
